From 50f052c6402bd25aba1e8c21d0cd8b1f3d3bac7a Mon Sep 17 00:00:00 2001
From: "Yong, Jonathan" <jonathan.yong@intel.com>
Date: Tue, 4 Aug 2020 04:25:42 +0000
Subject: [PATCH] linux.c: temporary workaround for broken upstream commit

| ../../git/grub-core/loader/i386/efi/linux.c: In function 'grub_cmd_linux':
| ../../git/grub-core/loader/i386/efi/linux.c:310:3: error: too many arguments to function 'grub_tpm_measure'
|   310 |   grub_tpm_measure (kernel, filelen, GRUB_BINARY_PCR, "grub_linuxefi", "Kernel");
|       |   ^~~~~~~~~~~~~~~~
| In file included from ../../git/grub-core/loader/i386/efi/linux.c:31:
| ../../git/include/grub/tpm.h:75:12: note: declared here
|    75 | grub_err_t grub_tpm_measure (unsigned char *buf, grub_size_t size,
|       |            ^~~~~~~~~~~~~~~~
| ../../git/grub-core/loader/i386/efi/linux.c:313:7: warning: implicit declaration of function 'grub_efi_secure_boot'; did you mean 'grub_efi_linux_boot'? [-Wimplicit-function-declaration]
|   313 |   if (grub_efi_secure_boot ())
|       |       ^~~~~~~~~~~~~~~~~~~~
|       |       grub_efi_linux_boot
| ../../git/grub-core/loader/i386/efi/linux.c:313:7: warning: nested extern declaration of 'grub_efi_secure_boot' [-Wnested-externs]

https://github.com/nicholasbishop/grub2/commit/94673c3e6a1cd8fc8fdb46faf9b4f8f912eae83a
https://github.com/nicholasbishop/grub2/commit/2045487bf6ed2a60b1f821b90b0cc2dac7ddb707

Upstream-status: Pending
Signed-off-by: Yong, Jonathan <jonathan.yong@intel.com>
---
 grub-core/loader/i386/efi/linux.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/grub-core/loader/i386/efi/linux.c b/grub-core/loader/i386/efi/linux.c
index 019a6f9dc..0af95a601 100644
--- a/grub-core/loader/i386/efi/linux.c
+++ b/grub-core/loader/i386/efi/linux.c
@@ -27,6 +27,7 @@
 #include <grub/lib/cmdline.h>
 #include <grub/efi/efi.h>
 #include <grub/efi/linux.h>
+#include <grub/efi/sb.h>
 #include <grub/cpu/efi/memory.h>
 #include <grub/tpm.h>
 #include <grub/safemath.h>
@@ -307,7 +308,7 @@ grub_cmd_linux (grub_command_t cmd __attribute__ ((unused)),
       goto fail;
     }
 
-  grub_tpm_measure (kernel, filelen, GRUB_BINARY_PCR, "grub_linuxefi", "Kernel");
+  grub_tpm_measure (kernel, filelen, GRUB_BINARY_PCR, "grub_linuxefi: Kernel");
   grub_print_error();
 
   if (grub_efi_secure_boot ())
-- 
2.25.1

